execution/gethexec/sequencer.go

```diff
@@ -332,12 +332,36 @@ func (c nonceFailureCache) Add(err NonceError, queueItem txQueueItem) {
 	}
 }
 
+type synchronizedTxQueue struct {
```

Comment:
Is it worth it to change this to something like `SynchronizedQueue[T any]` add it to containers package? Seems like it can be used in other places as well